sprout/appliances/models.py

```diff
@@ -671,9 +671,24 @@ def prolong_lease(self, time=60):
             appliance.prolong_lease(time=time)
 
     def kill(self):
+        save_lives = not self.fulfilled
         if self.appliances:
             for appliance in self.appliances:
-                Appliance.kill(appliance)
+                # The leased_until is reliable sign of whether the appliance was used
+                # Unless someone was messing with DB ;)
+                if appliance.ready and appliance.leased_until is None and save_lives:
```

Comment:
how do you mean that, the fulfilled property? I actually have to load it before the for loop because it *might* change during the loop.